run tests

itayd opened the pull request:
running on #23 (hehe), add run tests

tebeka commented on .github/workflows/ci-go.yml:
```diff
@@ -71,3 +71,18 @@ jobs:
+          go-version-file: go.mod
+
+      - name: CLI integration tests
+        run: make bin/ak test-runs
```
I prefer to have a single `ci` Makefile rule and have the logic of what to run in CI there instead of this YAML file.

itayd replied:
looking into it - might change in a future pr.

tebeka commented on cmd/ak/cmd/runtimes/run.go:
```diff
@@ -125,14 +126,18 @@ func init() {
 		Print: func(_ context.Context, _ sdktypes.RunID, msg string) {
-			fmt.Println(msg)
+			if !quiet {
+				fmt.Println(msg)
```
Why not log? (in human format)

itayd replied:
might append time and friends. we want to be able to compare it.

tebeka commented on tests/runs/load.txtar:
```diff
@@ -0,0 +1,24 @@
+["cat", "dog", "cow", "pig", "sheep", "chicken", "duck", "horse", "human"]
+["meow", "woof", "moo", "oink", "baa", "cluck", "quack", "neigh", "oi oi oi"]
```
:)

I sometime use [animal group names](https://arapahoelibraries.org/blogs/post/names-for-groups-of-animals/)

itayd replied:
no humans

itayd replied:
"Cats: a clowder, glaring, pounce, nuisance or clutter; Kittens: a litter or kindle; Wild cats: a destruction"

a destruction hahaha

tebeka commented on tests/runs/run.sh:
```diff
@@ -0,0 +1,18 @@
+
+AK="../../bin/ak"
+
+run() {
```
What is the value of this function?

itayd replied:
none what so ever. removed.

tebeka reviewed (commented):
I get errors when running `make test-race`.

```
2024/02/26 19:25:02 /home/miki/work/autokitteh/src/autokitteh/backend/internal/db/dbgorm/sessions.go:125 SQL logic error: no such table: sessions (1)
[0.276ms] [rows:0] SELECT * FROM `sessions` ORDER BY created_at desc
    ak_test.go:133: line 20: wait 5s for session s:00000000000000000000000000000008
    ak_test.go:135: error: session s:00000000000000000000000000000008 not done after 5s
        Events list:
        Error: list events: rpc: server error: db: SQL logic error: no such table: events (1) ([])
        ---
        Sessions list:
        Error: list sessions: rpc: db: SQL logic error: no such table: sessions (1) ([])

=== FAIL: systest TestSuite (12.80s)

DONE 231 tests, 2 failures in 18.315s
make: *** [Makefile:125: test-race] Error 1

```

itayd committed:
run locally or remotely; more run and build options (#23)

- building on #22 (sorry) , this also allows to run code out of session
either locally or remotely.
- allow running and building a txtar file
- support nested txtars in system test
- allow specifying `--test` to run, which will compare output with txtar
comment

itayd commented:
re https://github.com/autokitteh/autokitteh/pull/25#pullrequestreview-1901574358: @daabr is investigating.

itayd committed:
fix

itayd committed:
fix
